src/Native/configure.cmake

```diff
@@ -234,6 +234,22 @@ check_cxx_source_compiles(
 
 check_cxx_source_compiles(
     "
+    #include <netinet/tcp.h>
+    int main() { int x = TCP_ESTABLISHED; return x; }
+    "
+    HAVE_TCP_H_TCPSTATE_ENUM
+)
+
+check_cxx_source_compiles(
+    "
+    #include <netinet/tcp_fsm.h>
+    int main() { int x = TCPS_ESTABLISHED; return x; }
+    "
+    HAVE_TCP_FSM_H
+)
+
```

Comment:
Yup your approach is more clear.
`check_type_size` name is bit misleading, as the same macro is responds to different categories of checks. For instance, I used it above to test the presence of a struct in socket include files.